app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -104,6 +112,127 @@ export const exportedForTest = {
 };
 
 const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(superclass: TBase) => class extends superclass {
+    /**
+     * handleScheduledPosts: Handler responsible for the Create/Update operations occurring the SchedulePost table from the 'Server' schema
+     * @param {HandleScheduledPostsArgs} ScheduledPostsArgs
+     * @returns {Promise<ScheduledPostModel[]>}
+     */
+    handleScheduledPosts = async ({actionType, scheduledPosts, includeDirectChannelPosts, prepareRecordsOnly}: HandleScheduledPostsArgs): Promise<ScheduledPostModel[]> => {
+        const database: Database = this.database;
+        const scheduledPostsToDelete: ScheduledPostModel[] = [];
+        const scheduledPostsToCreateAndUpdate: ScheduledPostModel[] = [];
+
+        const currentTeamId = await getCurrentTeamId(database);
+
+        if (!scheduledPosts?.length) {
```

Comment:
By changing the condition in the if, you can get rid of the special case inside the if:

```suggestion
        if (!scheduledPosts?.length && actionType !== ActionType.SCHEDULED_POSTS.RECEIVED_ALL_SCHEDULED_POSTS) {
```

This way, the records to delete get calculated in whatever logic you have later for that purpose.

Reply:
I was talking about the outer if. So this will end up like...

````
if (!scheduledPosts?.length && actionType !== ActionType.SCHEDULED_POSTS.RECEIVED_ALL_SCHEDULED_POSTS) {
    logWarning('An empty or undefined "scheduledPosts" array has been passed to the handleScheduledPosts method');
    return [];
}

switch (actionType) {
...
```

Reply:
Got it, updated the code.